pkgs/by-name/ap/apt-mirror/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, perl
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "apt-mirror";
+  version = "0.5.4";
+
+  src = fetchFromGitHub {
+    owner = "apt-mirror";
+    repo = "apt-mirror";
+    rev = finalAttrs.version;
+    hash = "sha256-GNsyXP/O56Y+8QhoSfMm+ig5lK/K3Cm085jxRt9ZRmI=";
+  };
+
+  nativeBuildInputs = [ perl ];
+
+  installPhase = ''
+    runHook preInstall
+
+    # These commands are taken from the makefile
+    mkdir -p $out/{bin,share/man/man1,etc,var/spool/apt-mirror}
+    install -m 755 apt-mirror $out/bin/apt-mirror
+    pod2man apt-mirror > $out/share/man/man1/apt-mirror.1
+    install -m 644 mirror.list $out/etc/mirror.list
+    mkdir -p $out/var/spool/apt-mirror/{mirror,skel,var}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A tool that provides the ability to mirror any parts of apt sources";
+    homepage = "https://github.com/apt-mirror/apt-mirror";
+    changelog = "https://github.com/apt-mirror/apt-mirror/blob/${src.rev}/CHANGELOG";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ ];
```

Comment:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

`meta.maintainers` must be set

Reply:
Technically it is set - to an empty set. :)

On the other hand, I believe it can be adopted by another person, not necessarily the OC.

Reply:
as I wrote that I thought the same thing,
but surely the intention that it must be set to something that is not `empty list` or `null`

if thats indeed the case, we should update the README.md

Reply:
So, I can maintain this package, but I would like to know which is the correct behavior.

Reply:
@arthsmn you have created this PR to address https://github.com/NixOS/nixpkgs/issues/276763
but actually is have no experience with apt-mirror ? ( which is totally fine )

then we need to disambiguate what 
```
meta.maintainers must be set
```

means.

I always interpreted it as `maintainers` must be set to something other than empty list. After we find the answer  we should update documentation after this to remove ambiguation.

so whats the correct interpretation. maybe @Janik-Haag @natsukagami can help ?

Reply:
Yep, I don't have experience.

Reply:
That's fine. Let's just wait to get clarification about meta.maintainers

Reply:
I understand that the meta.maintainers of a new package should be set with a non-empty list, and I asked the documentation for the justification.

> meta.maintainers must be set for new packages.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

It certainly looks like meta.maintainers would be a good empty list.

> Maintainers must be set. This can be the package submitter or a community member that accepts taking up maintainership of the package.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#new-packages

This clearly states that an existing maintainer is required.

> Step3
> Some notes:
> Add yourself as the maintainer of the package.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package

It tells us that the submitter should be a maintainer.

Based on these considerations, my conclusion is this.
New packages must have at least one person assigned as meta.maintainers.
That is usually the submitter, but others can be assigned with permission.

Reply:
Thanks for the clarification!

Reply:
Yes new packages *MUST* have a maintainer. Packages might get an empty list of maintainers if people aren't careful while reviewing prs or if someone steps away from the Project or if the maintainer is inactive for multiple months.